NeoML/src/Dnn/Layers/FocalLossLayer.cpp

```diff
@@ -67,11 +67,18 @@ CFocalLossLayer::CFocalLossLayer( IMathEngine& mathEngine ) :
 	maxProbValue->GetData().SetValue( MaxProbValue );
 }
 
+void CFocalLossLayer::Reshape()
+{
+	CLossLayer::Reshape();
+	CheckArchitecture( inputDescs[1].GetDataType() == CT_Float, GetName(), "labels must be CT_Float" );
+}
+
 void CFocalLossLayer::BatchCalculateLossAndGradient( int batchSize, CConstFloatHandle data, int vectorSize,
 	CConstFloatHandle label, int labelSize, CFloatHandle lossValue, CFloatHandle lossGradient )
 {
-	NeoAssert( vectorSize == labelSize );
-	NeoAssert( labelSize >= 2 );
+	CheckArchitecture( labelSize == vectorSize, GetName(), "the labels dimensions should be equal to the first input dimensions" );
```

Comment:
What does `first` mean here? Because in NeoML order `batchSize` goes _prior_ to `vectorSize`. M.b. just use vector size?

Reply:
Why not check it in `::Reshape()`?